themes/default/content/challenge/holiday-shopping.md

```diff
@@ -0,0 +1,268 @@
+
+```shell
+pulumi config set orgID [value]
+pulumi config set dbUser [value]
+pulumi config set dbPassword [value] --secret
+```
+
+#### Step 4. Pass database URL as environment variable to backend container
+
+Before you update the stack, you need to pass the URL of the database to the backend container so it can connect to it. Add the highlighted into the FargateService code.
+
+```python {.line-numbers hl_lines=["32-37"]}
+service = awsx.ecs.FargateService(
+    "grocery-service",
+    cluster=cluster.arn,
+    assign_public_ip=True,
+    task_definition_args=awsx.ecs.FargateServiceTaskDefinitionArgs(
+        containers={
+            "front": awsx.ecs.TaskDefinitionContainerDefinitionArgs(
+                image=frontend_image.image_uri,
+                cpu=cpu,
+                memory=memory,
+                essential=True,
+                port_mappings=[awsx.ecs.TaskDefinitionPortMappingArgs(
+                    container_port=container_port,
+                    target_group=lb.default_target_group,
+                )],
+                environment=[{
+                    # Unused unless running dev server
+                    "name":"VITE_BACKEND_URL",
+                    "value":"http://localhost:8000"
+                },
+                ],
+            ),
+            "back": awsx.ecs.TaskDefinitionContainerDefinitionArgs(
+                image=backend_image.image_uri,
+                cpu=cpu,
+                memory=memory,
+                essential=True,
+                port_mappings=[awsx.ecs.TaskDefinitionPortMappingArgs(
+                    container_port=8000,
+                    host_port=8000
+                )],
+                environment=[{
+                    "name":"DATABASE_URL",
+                    "value":Output.format("mongodb+srv://{0}:{1}@{2}", db_username, db_password,
+                                Output.all(mongo_cluster.srv_address).apply(lambda v: v[0].split("//"))[1])
+                },
+                ],
+            ),
+        }
+    ),
+    desired_count=1
+)
+```
+
+#### Step 5. Validate endpoints
+
+Run `pulumi up` and select yes to update the stack. Go to the `app url` export to validate that the endpoints of your deployed infrastructure is working. You can now add, delete, and check off items because the application is now able to connect to your newly created Atlas database.
+
+![alt_text](/images/challenge/grocery_list.png "grocery list app")
+
+#### Congratulations
+
+Congratulations!. You've completed this Pulumi Challenge. If you'd like to tear down all of these resources, run `pulumi destroy -y`. Otherwise, have fun playing around with your infrastructure stack and add whatever you like! 🙂
```

Comment:
```suggestion
Congratulations! You've completed this Pulumi Challenge. If you'd like to tear down all of these resources, run `pulumi destroy -y`. Otherwise, have fun playing around with your infrastructure stack and add whatever you like! 🙂
```